youtube_dl/extractor/twitter.py

```diff
@@ -280,7 +280,7 @@ def _real_extract(self, url):
 
 class TwitterIE(InfoExtractor):
     IE_NAME = 'twitter'
-    _VALID_URL = r'https?://(?:www\.|m\.|mobile\.)?twitter\.com/(?:i/web|(?P<user_id>[^/]+))/status/(?P<id>\d+)'
+    _VALID_URL = r'https?://(?:www\.|m\.|mobile\.)?twitter\.com/(?:i/web|-|(?P<user_id>[^/]+))/status/(?P<id>\d+)'
```

Comment:
You must fix user_id extraction instead of this ugly hack.

Reply:
Okay, should I modify here then?
https://github.com/ytdl-org/youtube-dl/blob/master/youtube_dl/extractor/twitter.py#L444